pkg/controllers/staticroute/staticroute_controller.go

```diff
@@ -0,0 +1,285 @@
+
+	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
+		log.Error(err, "unable to fetch static route CR", "req", req.NamespacedName)
+		return common.ResultNormal, client.IgnoreNotFound(err)
+	}
+
+	// Since StaticRoute service can only be activated from NSX 4.1.0 onwards,
+	// So need to check NSX version before starting StaticRoute reconcile
+	if !r.Service.NSXClient.NSXCheckVersionForStaticRoute() {
+		err := errors.New("NSX version check failed, StaticRoute feature is not supported")
+		updateFail(r, &ctx, obj, &err)
+		// if NSX version check fails, it will be put back to reconcile queue and be reconciled after 5 minutes
+		return ResultRequeueAfter5mins, nil
+	}
+
+	if obj.ObjectMeta.DeletionTimestamp.IsZero() {
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeStaticRoute)
+		if !controllerutil.ContainsFinalizer(obj, commonservice.StaticRouteFinalizerName) {
+			controllerutil.AddFinalizer(obj, commonservice.StaticRouteFinalizerName)
+			if err := r.Client.Update(ctx, obj); err != nil {
+				log.Error(err, "add finalizer", "staticroute", req.NamespacedName)
+				updateFail(r, &ctx, obj, &err)
+				return common.ResultRequeue, err
+			}
+			log.V(1).Info("added finalizer on staticroute CR", "staticroute", req.NamespacedName)
+		}
+
+		if err := r.Service.CreateorUpdateStaticRoute(obj); err != nil {
+			log.Error(err, "operate failed, would retry exponentially", "staticroute", req.NamespacedName)
+			updateFail(r, &ctx, obj, &err)
+			// TODO: if error is not retriable, not requeue
+			return common.ResultRequeue, err
+		}
+		updateSuccess(r, &ctx, obj)
+	} else {
+		if controllerutil.ContainsFinalizer(obj, commonservice.StaticRouteFinalizerName) {
+			metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteTotal, common.MetricResTypeStaticRoute)
+			// TODO, update the value from 'default' to actual value， get OrgID, ProjectID, VPCID depending on obj.Namespace from vpc store
+			if err := r.Service.DeleteStaticRoute("default", "default", "default", string(obj.UID)); err != nil {
+				log.Error(err, "delete failed, would retry exponentially", "staticroute", req.NamespacedName)
+				deleteFail(r, &ctx, obj, &err)
+				return common.ResultRequeue, err
+			}
+			controllerutil.RemoveFinalizer(obj, commonservice.StaticRouteFinalizerName)
+			if err := r.Client.Update(ctx, obj); err != nil {
+				deleteFail(r, &ctx, obj, &err)
+				return common.ResultRequeue, err
+			}
+			log.V(1).Info("removed finalizer", "staticroute", req.NamespacedName)
+			deleteSuccess(r, &ctx, obj)
+		} else {
+			// only print a message because it's not a normal case
+			log.Info("finalizers cannot be recognized", "staticroute", req.NamespacedName)
+		}
+	}
+
+	return common.ResultNormal, nil
+}
+
+func (r *StaticRouteReconciler) setStaticRouteReadyStatusTrue(ctx *context.Context, static_route *v1alpha1.StaticRoute) {
+	newConditions := []v1alpha1.StaticRouteCondition{
```

Comment:
checking on staticroute define, why define a new staticroutecondition instead of using condition directly.

Reply:
please ignore this comment, i got the reason...